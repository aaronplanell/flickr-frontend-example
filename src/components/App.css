@charset "utf-8";

html {
  height: 97%;
}

body {
  height: 100%;
	font: 24px Open Sans;
	background: white;
}

#main {
	min-height: 400px;
  height: 100%;
	margin: 0px;
	padding: 0px;
	display: 	-webkit-flex;
	display:         	flex;
				-webkit-flex-flow: row;
				flex-flow: row;
}

#main > nav {
  height: 100%;
	margin: 4px;
	padding: 5px;
	border: 1px solid wheat;
	border-radius: 7pt;
	background: ivory;
				-webkit-flex: 1 6 10%;
						flex: 1 6 10%;
				-webkit-order: 1;
	order: 1;
	font-size: 16px;
	font-style: center;
	text-align:center;
  max-width: 150px;
}

#main > section, article {
	margin: 4px;
	padding: 5px;
	border: 1px solid wheat;
	border-radius: 7pt;
	min-height: 100px;
	background: ivory;
				-webkit-flex: 3 1 75%;
						flex: 3 1 75%;
				-webkit-order: 2;

	font-size: 18px;
	order: 2;
  height: 88%;
}

#main > aside {
  height: 100%;
	margin: 4px;
	padding: 5px;
	border: 1px solid wheat;
	border-radius: 7pt;
	background: ivory;
				-webkit-flex: 1 6 15%;
						flex: 1 6 15%;
				-webkit-order: 3;
	order: 3;
	font-size: 16px;
	font-style: center;
	text-align:center;
  max-width: 150px;
}

#main > aside > img {
	display: block;
	margin: auto;
	width: 90%;
	padding: 5%;
}

ul {
	list-style-type: none;
	margin: 0px;
	padding: 0px;
	font-size: 16px;
	font-weight: bold;
}

ul > a {
	color: #000;
	text-decoration: none;
}

li {
	padding-left: 10px;
}

ul > a:hover {
	color: #FFF;
}

iframe {
	display: inline;
	border: 1px solid white;
	background-color: white;
	width: 300px;
	height: 196px;
}

footer {
  font-size: 12px;
}

/* Link as a button */
a.btn {
	display: 	-webkit-flex;
	display:         	flex;
				-webkit-flex-flow: row;
				flex-flow: row;
	color: #666;
	background-color: #eee;
	letter-spacing: 2px;
	font-size: 12px;
	padding: 10px 10px;
	border-radius: 5px;
	-moz-border-radius: 5px;
	-webkit-border-radius: 5px;
	border: 1px solid rgba(0,0,0,0.3);
	border-bottom-width: 3px;
	min-width: 80px;
  text-decoration: none
}

a.btn:hover {
	background-color: #e3e3e3;
	border-color: rgba(0,0,0,0.5);
}

a.btn:active {
	background-color: #CCC;
	border-color: rgba(0,0,0,0.9);
}

/* Too narrow to support three columns */
@media all and (max-width: 768px) {
	#main, #page {
		-webkit-flex-flow: column;
		flex-flow: column;
	}

	#main > section, #main > article, #main > nav, #main > aside {
	/* Return them to document order */
		-webkit-order: 0;
		order: 0;
	}

	#main > nav, #main > aside, header, footer {
		min-height: 40px;
		max-height: 40px;
	}

	#main > section {
		height: 100%;
	}

	#main > nav {
		width: 100%;
		height: 30px;
	}

	ul {
		display: inline-flex;
	}

}
