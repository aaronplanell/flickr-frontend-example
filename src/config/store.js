import { createStore } from 'redux';
import rootReducer from '../reducers';
import getMiddlewares from './middlewares';

export default function configureStore(preloadedState) {
  return createStore(rootReducer, preloadedState, getMiddlewares());
}
