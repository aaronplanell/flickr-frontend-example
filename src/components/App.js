import React, { Component } from 'react';
import { connect } from 'react-redux';

// CSS
import './App.css';

// Components
import Navigation from './sections/Navigation';
import Header from './sections/Header';
import Article from './sections/Article';
import Footer from './sections/Footer';
import Aside from './sections/Aside';

// Actions
import {
  fetchListOfPhotosets,
  selectPhotoset,
  fetchPhotosByPhotoset,
  getSizesOfAllPhotos,
  fetchSizesOfAPhoto,
  selectViewSize,
  selectPhoto
} from '../actions';

// selectorOfPhotosBySize
import { selectorOfPhotosBySize } from '../reducers/photos';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      isListOfPhotosetsLoading: false
    }
    fetchListOfPhotosets();
  }

  componentDidMount() {
    const { isListOfPhotosetsLoaded } = this.state;
    const { fetchListOfPhotosets } = this.props;
    if (!isListOfPhotosetsLoaded) {
      this.setState({...this.state, isListOfPhotosetsLoaded: true});
      fetchListOfPhotosets();
    }
  }

  componentWillReceiveProps(nextProps) {
    const { params, photos, getSizesOfAllPhotos, fetchSizesOfAPhoto } = nextProps;
    const { updateSizesOfPhotos } = params;
    if (updateSizesOfPhotos) {
      photos.map( (photo) => {
        const { id } = photo;
        return fetchSizesOfAPhoto(id);
      });
      getSizesOfAllPhotos(false);
    };
  }

  render() {
    const {
      alert,
      params,
      photosets,
      selectPhotoset,
      fetchPhotosByPhotoset,
      getSizesOfAllPhotos,
      selectedPhotos,
      selectViewSize,
      selectPhoto
    } = this.props;
    const { currentViewSize, selectedPhoto } = params;

    return (
      <div id="main" className="App" style={{height: '100%'}}>
        <Navigation
          selectViewSize={selectViewSize}
          currentViewSize={currentViewSize}
        />
        <Aside
          photosets={photosets}
          selectPhotoset={selectPhotoset}
          fetchPhotosByPhotoset={fetchPhotosByPhotoset}
          getSizesOfAllPhotos={getSizesOfAllPhotos}
          selectPhoto={selectPhoto}
        />
        <section style={{height: '100%'}}>
          <Header />
          <Article
            alert={alert}
            selectedPhotos={selectedPhotos}
            currentViewSize={currentViewSize}
            selectPhoto={selectPhoto}
            selectedPhoto={selectedPhoto}
          />
          <Footer />
        </section>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    ...state,
    selectedPhotos: selectorOfPhotosBySize(state.photos, state.params.currentViewSize)
  }
}

export default connect(mapStateToProps, {
  fetchListOfPhotosets,
  selectPhotoset,
  fetchPhotosByPhotoset,
  getSizesOfAllPhotos,
  fetchSizesOfAPhoto,
  selectViewSize,
  selectPhoto
})(App);
