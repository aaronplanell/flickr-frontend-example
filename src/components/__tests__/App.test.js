import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../config/mockStore';
import App from '../App';

jest.mock('../../actions', () => 'actions' );

jest.mock('../sections/Navigation', () => {
  return (props) => <div>{props.children}</div>;
});

jest.mock('../sections/Aside', () => {
  return (props) => <div>{props.children}</div>;
});

jest.mock('../sections/Article', () => {
  return (props) => <div>{props.children}</div>;
});

jest.mock('../sections/Article/Photo', () => {
  return (props) => <div>{props.children}</div>;
});

it('App Renders OK', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <App fetchListOfPhotosets={() => {return null}}/>
    </Provider>,
    div);
});

it('App Snapshot matchs', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <App fetchListOfPhotosets={() => {return null}}/>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
