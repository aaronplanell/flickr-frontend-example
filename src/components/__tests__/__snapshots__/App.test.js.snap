exports[`test App Snapshot matchs 1`] = `
<div
  className="App"
  id="main"
  style={
    Object {
      "height": "100%",
    }
  }>
  <div />
  <div />
  <section
    style={
      Object {
        "height": "100%",
      }
    }>
    <header>
      <strong>
        Flickr Frontend Example
      </strong>
    </header>
    <div />
    <footer>
      <a
        href="mailto:aaronplanell@gmail.com"
        style={
          Object {
            "paddingRight": "5px",
          }
        }>
        aaronplanell@gmail.com
      </a>
      ···
      <a
        href="https://github.com/aaronplanell/flickr-frontend-example"
        style={
          Object {
            "paddingLeft": "5px",
          }
        }
        target="_blank">
        https://github.com/aaronplanell/flickr-frontend-example
      </a>
    </footer>
  </section>
</div>
`;
